### PR TITLE
Implementation of EmailService

### PR DESCRIPTION
**Pull request by fabriciodigennaro**
## Trello ticket
[Ticket](/58-implement-email-sender-service)

## Description of changes

- Implementation of JavaEmailSender
- Created EmailService
- Added Unit test
- Added architecture tests

### CODE REVIEW

File: src/test/java/com/parkingapp/notificationservice/application/sendemailnotification/SendEmailNotificationUseCaseTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    @Test
+    void shouldSendAnEmailNotification() {
+        // GIVEN
+        when(emailTemplateRepository.getEmailTemplateById(templateId)).thenReturn(emailTemplate);
+
+        // WHEN
+        useCase.execute(userId, templateId);
+
+        // THEN
+        verify(emailTemplateRepository).getEmailTemplateById(templateId);
```
Comment (solblack) [inline src/test/java/com/parkingapp/notificationservice/application/sendemailnotification/SendEmailNotificationUseCaseTest.java:39]: I'm missing verification of EmailService send method invocation

File: src/main/java/com/parkingapp/notificationservice/infrastructure/email/JmsEmailService.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    private final String sender;
+
+    public JmsEmailService(JavaMailSender javaMailSender, String sender) {
+        this.javaMailSender = javaMailSender;
+        this.sender = sender;
+    }
+
+    @Override
+    public void send(EmailNotification emailNotification) {
```
Comment (solblack) [inline src/main/java/com/parkingapp/notificationservice/infrastructure/email/JmsEmailService.java:24]: missings tests

File: src/test/java/com/parkingapp/notificationservice/infrastructure/email/JmsEmailServiceIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    @Autowired
+    private EmailService emailService;
+
+    @AfterEach
+    void tearDown() {
+        greenMail.reset();
+    }
+
+    @Test
+    void testSendEmail() throws Exception {
```
Comment (solblack) [inline src/test/java/com/parkingapp/notificationservice/infrastructure/email/JmsEmailServiceIntegrationTest.java:37]:
```suggestion
    void shouldSendEmail() throws Exception {
```

**Review by solblack — APPROVED** [review]

File: src/test/java/com/parkingapp/notificationservice/infrastructure/email/JmsEmailServiceIntegrationTest.java (hunk 3 above)
Comment (fabriciodigennaro) [reply src/test/java/com/parkingapp/notificationservice/infrastructure/email/JmsEmailServiceIntegrationTest.java]: 👍🏻

**Pull request merged**